gh-skeleton

```diff
@@ -0,0 +1,254 @@
+
+Usage:
+  gh skeleton (-h | --help)
+  gh skeleton list [--src-org <name>]
+  gh skeleton clone [options] <parent-repo-name> <new-repo-name>
+
+Options:
+  -c --change-dir <dir> Create clone in this directory.
+  -d --dest-org <name>  Organization to create clone into [default: ${DEST_ORG}].
+  -h --help             Show this message.
+  -s --src-org <name>   Organization to search for skeletons [default: ${SRC_ORG}].
+END_OF_LINE
+)
+
+print_available_skeletons() {
+  src_org=$1
+  QUERY="
+  query {
+    search(query: \"org:$src_org topic:skeleton archived:false\", type: REPOSITORY, first: 100) {
+      repositoryCount
+      edges {
+        node {
+          ... on Repository {
+            name
+            description
+          }
+        }
+      }
+    }
+  }
+  "
+
+  # shellcheck disable=2016
+  # Disable single quote expansion warning as we want to suppress
+  # parameter expansion in a Go template.
+  TEMPLATE='
+    {{- range $repo := .data.search.edges -}}
+      {{- printf "%-25s\t%s\n" $repo.node.name $repo.node.description -}}
+    {{- end -}}
+  '
+
+  echo "Available skeletons in $src_org:"
+  echo
+  exec gh api graphql \
+    --raw-field query="${QUERY}" --paginate \
+    --template="${TEMPLATE}" | sort
+}
+
+clone_repo() {
+  src_repo=$1
+  dest_repo=$2
+  src_org=$3
+  dest_org=$4
+
+  # Before we continue check to see if the user is logged in. If there is no
+  # currently authenticated user gh will ask the user to run the "gh auth login"
+  # command and exit with return code 4.  This will cause this script to halt.
+  username=$(gh api user --jq '.login')
+
+  # If we got here we know we have an authenticated user.
+  log_ok "You are logged into GitHub as ${username}."
+
+  log_info "Creating ${dest_org}/${dest_repo} from skeleton ${src_org}/${src_repo}."
+
+  log_info "Cloning skeleton remote repository to the new local repository."
+  git clone --origin "${src_repo}" git@github.com:"${src_org}/${src_repo}".git "${dest_repo}"
+
+  cd "${dest_repo}"
+
+  log_info "Disabling pushing to the upstream (parent) repository."
+  git remote set-url --push "${src_repo}" no_push
+
+  log_info "Creating the lineage.yml file."
+  cat << END_OF_LINE > .github/lineage.yml
+---
+lineage:
+  skeleton:
+    remote-url: https://github.com/${src_org}/${src_repo}.git
+version: '1'
+END_OF_LINE
+
+  log_info "Staging modified files."
+  git add --verbose .
+
+  log_info "Committing staged files to the develop branch."
+  git commit --message "Add lineage configuration"
+
+  log_info "Creating first-commits branch."
+  git checkout -b first-commits
+
+  log_info "Searching and replacing repository name in source files."
+  find . \( ! -regex '.*/\.git/.*' \) -type f -exec \
+    perl -pi -e "s/${src_repo}/${dest_repo}/g" {} \;
+
+  log_info "Staging modified files."
+  git add --verbose .
+
+  log_info "Committing staged files to the first-commits branch."
+  git commit --message "Rename repository references after clone"
```

Comment:
I like the latter (all `clone`-related housekeeping is pushed to `develop`, then create the `first-commits` branch from that) since that's the way we've been doing it and we haven't had any issues.

Reply:
Now that I've gone back and tested it... I realize there was a reason I was putting the lineage configuration into `develop` and the search and replace changes into the `first-commits` branch:

This gives `first-commits` a commit, which allows the PR that is opened at the end of the script to succeed.  The way it is now, `first-commits` has no commits, and the PR creation fails as a noop.

Reply:
The empty PR will suffice for now.  Most people will understand what is going on when they see the resulting PR page.  

@mcdonnnj Let's get this resolved so I can move on with #3.